.github/workflows/run_tests.yml

```diff
@@ -16,8 +16,13 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ["3.6", "3.7"]
-        pytorch-version: ["1.2"]
+        python-version: ["3.8", "3.12"]
+        pytorch-version: ["1.4", "2.3"]
+        exclude:
+          - python-version: "3.8"
+            pytorch-version: "2.3"
+          - python-version: "3.12"
+            pytorch-version: "1.4"
```

Comment:
Can you give reasons on the versions here? E.g., why Python 3.12 and `PyTorch` 2.3 are explicitly tested against here, etc.

Reply:
Torch 1.4 is the oldest available torch on PyPI for python 3.8, and torch 2.3 is the newest for python 3.12. I want the newest combination tested because that's what newcomers (like me) tend to have when they install their environment. The oldest combination is there to have some confidence in backward compatibility. 

The exclude there is because there is no torch 1.4 on python 3.12. After another check I found torch 2.3 is available on 3.8, should we include that as well?

Reply:
These are updated.